main.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+class HashTable {
+  constructor() {
+    this.values = {};
+    this.length = 0;
+    this.size = 0;
+  }
+
+  addOne(elem) {
+    const hash = ++this.size;
+    if (!this.values.hasOwnProperty(hash)) {
+      this.values[hash] = {};
+    }
+    for (const key in elem) {
+      if (!this.values[hash].hasOwnProperty(key)) {
+        this.length++;
+      }
+      this.values[hash][key] = elem[key];
+    }
+  }
+
+  addMany(arr) {
+    for (const item of arr) {
+      this.addOne(item);
+    }
+  }
+
+  searchByValue(value) {
+    const find = {};
+    if (typeof value === 'string') {
+      for (const hash in this.values) {
+        for (const item in this.values[hash]) {
```

Comment:
- for.of and simple c-for is better then for..in
- you can avaid nested loops here